.golangci-maximal.yml

```diff
@@ -0,0 +1,183 @@
+linters:
+  disable-all: true
+  enable:
+    - copyloopvar
+    - cyclop
+    - depguard
+    - dupword
+    - errcheck
+    - errname
+    - errorlint
+    - exhaustruct
+    - exptostd
+    - funlen
+    - gci
+    - gocheckcompilerdirectives
+    - gocognit
+    - gocritic
+    - gocyclo
+    - godot
+    - goimports
+    - govet
+    - iface
+    - importas
+    - ineffassign
+    - intrange
+    - maintidx
+    - misspell
+    - nestif
+    - nilnesserr
+    - nolintlint
+    - predeclared
+    - recvcheck
+    - revive
+    - staticcheck
+    - testifylint
+    - unconvert
+    - unparam
+    - unused
+    - usestdlibvars
+    - usetesting
+  fast: false
+
+linters-settings:
+  depguard:
+    rules:
+      # This is just an example of a rule you could create, though 
+      forbid-some-pacakge:
+        list-mode: Lax
+        files:
+          - "**/*.go"
+        deny:
+          pkg: "github.com/some/package"
+          desc: "Don't use some/package. It's got issues."
+
+  errcheck:
+    check-type-assertions: true
+
+  gci:
+    sections:
+      - standard
+      - default
+
+  # This configure gocritic so that it is _only_ used to run ruleguard. It's default checks include
+  # some that can be a bit annoying, like insisting on using `switch` instead of if/else all over
+  # the place.
+  gocritic:
+    disable-all: true
+    enabled-checks:
+      - ruleguard
+    settings:
+      ruleguard:
+        rules: "${configDir}/gorules/*.go"
+        # This causes a warning when rules cannot be compiled.
+        failOn: dsl
+
+
+  govet:
+    # This is not part of the default checks, but it's useful for finding bugs and generally
+    # confusing code.
+    shadow: true
+
+  importas:
+    no-extra-aliases: true
+    alias:
+      - pkg: github.com/google/uuid
+        alias: guuid
+
+  misspell:
+    locale: US
+    extra-words:
+      - typo: "cancelation"
+        correction: "cancellation"
+      - typo: "cancelations"
+        correction: "cancellations"
+      - typo: "cancelling"
+        correction: "canceling"
+      - typo: "cancelled"
+        correction: "canceled"
+
+  recvcheck:
+    exclusions:
```

Comment:
I kinda of wish we used value types more often. Certainly use-case dependent. Does this effectively prohibit that without nolint directives?

Reply:
This linter complains if you mix value & pointer receivers for the same type. If you're always consistent, it doesn't care which one you use.